docs/en/week12/12-1.md

```diff
@@ -0,0 +1,220 @@
+<b>Figure 6:</b> Combine target hidden with source vector
+</center>
+
+<br/>
+
+Alignment is learnt implicitly in seq2seq model. All tokens can be processed in parallel efficiently with CNNs or Transformers.
+
+### Test NMT
+After we have the translation model, and know how good they are doing on the training set, we will want to test to see how the model will actually generate translations from the source sentences.
+
+Beam Decoding is used to search in the space of $y$ according to
+
+$$ {\operatorname{argmax}} \log p(\vec{y} \mid \vec{x};\theta) $$
+
+Each potential choice (the number of words in vocabulary) has a probability score. At every step, beam search select the top K scoring among all the branches (maintain queue with k top scoring paths), then expand each of them and retain the top k scoring paths. As illustrated by the example below:
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-1/figure7.png" style="background-color:#DCDCDC;"><br>
+<b>Figure 7:</b> Beam Search
+</center>
+
+Let $k = 2$
+1. Start from symbol /s, pick the top 2 scoring from all the possible choices (Life, Today)
+2. Continue from Life and Today, expand each of them and calculate the path score back to the start symbol, pick the top 2 scoring path ("Like was" and "Like is") 
+3. So on and so for, keep proceeding until hit the end of the sentence
+4. At the very last step, select the highest scoring path
+
+Beam search is a greedy procedure and it is very effective in practice. There is a trade-off between computational cost and approximation error (the larger the value of K, the better the approximation and the higher the computational cost). 
+
+Issues of Beam Search:
+Beam search will always select the highest scoring path. Thus, the solution will be biased because of such strategy. It doesn't handle uncertainty well. 
+
+Other decoding methods: sampling, top-k sampling, generative and discriminative reranking 
+
+### NMT Training & Inference Summary
+
+Training: predict one target token at the time and minimize cross-entropy loss
+
+Inference: find the most likely target sentence (approximately) using beam search
+
+Evaluation: compute BLEU on hypothesis returned by the inference procedure
+
+We first compute the geometric average of the modified n-gram precisions, $p_n$
+
+$$p_n = \frac{\sum_{\text{generated sentences}} \sum_{\text{ngrams}} \text{Clip(Count(ngram matches))}}{\sum_{\text{generated sentences}} \sum_{\text{ngrams}} \text{Count(ngram)}}$$
+
+Candidate translations longer than their references are already penalized by the modified n-gram precision measure: there is no need to penalize them again. Consequently, we introduce a multiplicative brevity penalty factor, $BP$. With $BP$ in place, a high-scoring candidate translation must now match the reference translations in length, in word choice, and in word order.
+
+$$BLEU = BP \exp ({\sum_{n=1}^{N} \frac{1}{N} \log \ p_n})$$
```

Comment:
Please, don't use multi-letter names, this is not math at all.
`$S_{\text{BLEU}}$` for BLEU score could be an option.
Again, fix the B * P thing.
Parenthesis should have a `\` in front.

Reply:
I apologise, this was not my part but I will fix this ASAP.

Changes I am making:
- Changing BLEU score to `s_{\text{BLEU}}` and BP factor to `f_{BP}`
- `\` is not working, so I am using `\left(` and `\right)` instead

If these changes are okay, I will incorporate them.

Reply:
Go ahead.